### PR TITLE
Integrate Instructor Library for easy LLM interaction

### LINKED ISSUE #4: Use Instructor to provide easy LLM interaction
https://github.com/jxnl/instructor/
https://python.useinstructor.com/examples/knowledge_graph/

### PR DESCRIPTION
**Pull request by anubhuti24**
closes #4

### CODE REVIEW

File: libsql_graph_db/natural.py [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+from fastapi import FastAPI
+from models import KnowledgeGraph  # type: ignore
+
+load_dotenv()
+client = instructor.patch(OpenAI(api_key=os.getenv("OPEN_API_KEY")))
+
+app = FastAPI()
+
+
+@app.get("/graph/insert", response_model=KnowledgeGraph)
```
Comment (sutyum) [inline libsql_graph_db/natural.py:18]: no need of exposing a REST api, this is just a python library

**Review by sutyum — CHANGES_REQUESTED** [review]

File: libsql_graph_db/models.py [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+    node_identifier: str = Field(
+        ..., description="Unique Identifier which defines the node."
+    )
+    body: str = Field(
+        ..., description="Content or information associated with the node."
+    )
+    label: Optional[str] = Field(
+        None, description="Label or name associated with the node."
+    )
+    color: Optional[str] = Field(
```
Comment (sutyum) [inline libsql_graph_db/models.py:19]: color isnt needed

File: libsql_graph_db/models.py [hunk 3]
```diff
@@ -1,3 +1,40 @@
...
+class Edge(BaseModel):
+    id: int = Field(..., description="Unique identifier for the edge.")
+    source: int = Field(
+        ..., description="ID of the source node from which the edge originates."
+    )
+    target: int = Field(
+        ..., description="ID of the target node to which the edge points."
+    )
+    label: str = Field(..., description="Label or property associated with the edge.")
+    color: Optional[str] = Field(
```
Comment (sutyum) [inline libsql_graph_db/models.py:33]: color isnt needed

File: libsql_graph_db/models.py [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
+
+
+class Node(BaseModel):
+    node_identifier: str = Field(
+        ..., description="Unique Identifier which defines the node."
+    )
+    body: str = Field(
+        ..., description="Content or information associated with the node."
+    )
+    label: Optional[str] = Field(
```
Comment (sutyum) [inline libsql_graph_db/models.py:16]: label shouldnt be optional

**Review by sutyum — CHANGES_REQUESTED** [review]

File: libsql_graph_db/models.py [hunk 5]
```diff
@@ -1,3 +1,40 @@
 """
 Pydantic models for pythonic API to the graph db
 """
+
+from pydantic import BaseModel, Field
+from typing import Optional, List
+
+
+class Node(BaseModel):
```
Comment (sutyum) [inline libsql_graph_db/models.py:9]: NOTE: these types will also be used in `graph.py` when wrapping the functions provided by `database.py` into a higher level API

**Review by sutyum — CHANGES_REQUESTED** [review]

File: libsql_graph_db/natural.py [hunk 6]
```diff
@@ -1,3 +1,35 @@
...
+            {
+                "role": "user",
+                "content": f"Help me describe this user query as a detailed knowledge graph with proper, meaningful and unique relationships and edges: {query}",
+            },
+        ],
+        response_model=KnowledgeGraph,
+    )
+
+    for node in kg.nodes:
+        atomic(add_node({"body": node.body}, node.node_identifier), db_url, auth_token)
```
Comment (sutyum) [inline libsql_graph_db/natural.py:33]: the relationships?

**Review by sutyum — CHANGES_REQUESTED** [review]

File: libsql_graph_db/natural.py [hunk 7]
```diff
@@ -1,3 +1,35 @@
-"""
-Use models.py and Instructor to provide a natural language interface to the graph db
-"""
+import os
+import instructor
+from openai import OpenAI
+from dotenv import load_dotenv
+from models import KnowledgeGraph  # type: ignore
```
Comment (sutyum) [inline libsql_graph_db/natural.py:5]: `# type: ignore` this should not be required for files written by us, if you are needing this - that means some types are probably missing in that file. Look at the mypy errors and fix the typing issues there

File: libsql_graph_db/natural.py [hunk 8]
```diff
@@ -1,3 +1,35 @@
...
+from database import initialize, add_node, atomic  # type: ignore
```
Comment (sutyum) [inline libsql_graph_db/natural.py:6]: `# type: ignore` this should not be required for files written by us, if you are needing this - that means some types are probably missing in that file. Look at the mypy errors and fix the typing issues there

**Review by sutyum — CHANGES_REQUESTED** [review]

File: libsql_graph_db/natural.py (hunk 8 above)
Comment (anubhuti24) [reply libsql_graph_db/natural.py]: Done

File: libsql_graph_db/natural.py (hunk 7 above)
Comment (anubhuti24) [reply libsql_graph_db/natural.py]: Done

File: libsql_graph_db/natural.py (hunk 6 above)
Comment (anubhuti24) [reply libsql_graph_db/natural.py]: Added relationships

File: libsql_graph_db/models.py (hunk 4 above)
Comment (anubhuti24) [reply libsql_graph_db/models.py]: Updated models

**Review by sutyum — CHANGES_REQUESTED** [review]
add example showing this feature being used

**Review by sutyum — APPROVED** [review]

**Pull request merged**